clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -209,15 +213,21 @@ public NetworkClientDelegate.UnsentRequest toUnsentRequest() {
                 requestTopicDataMap.put(topicPartition.topic(), topic);
             }
 
-            OffsetCommitRequest.Builder builder = new OffsetCommitRequest.Builder(
-                    new OffsetCommitRequestData()
-                            .setGroupId(this.groupId)
-                            .setGenerationIdOrMemberEpoch(generation.generationId)
-                            .setMemberId(generation.memberId)
-                            .setGroupInstanceId(groupInstanceId)
-                            .setTopics(new ArrayList<>(requestTopicDataMap.values())));
+            OffsetCommitRequestData data = new OffsetCommitRequestData()
+                    .setGroupId(groupId)
+                    .setGenerationIdOrMemberEpoch(membershipManager.memberEpoch())
```

Comment:
We end up taking the epoch, memberId, and groupInstanceId directly from the `membershipMgr`, even though they are defined as local variables (unused) in this `OffsetCommitRequestState` class. In the case of the groupId though, we do use the local var. Let's review and make it consistent.